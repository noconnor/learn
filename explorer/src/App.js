import React, { Component } from 'react';

import 'App.css';
import ExplorerForm from 'components/Explorer/Form'
import Menu from 'components/Explorer/Greetings'
import owl from 'owl.png';

class App extends Component {
  render() {
    if (this.props.authState === "signedIn") {
      return (
        <div className="App">
          <Menu {...this.props} />
          <header className="App-header">
              <img src={owl} className="App-logo" alt="logo" />
              <p />
              <ExplorerForm {...this.props}/>
          </header>
        </div>
      );
    } else {
      return null;
    }
  }
}

export default App;