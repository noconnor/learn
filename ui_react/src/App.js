import React, { Component } from 'react';

import './App.css';
// import ExplorerForm from './components/Explorer/Form'
import LoginForm from './components/Login/Form'
import owl from './owl.png';

import Amplify from 'aws-amplify';

Amplify.configure({
    Auth: {        
        region: 'us-east-1',
        userPoolId: 'us-east-1_ad7bv0cEJ',
        userPoolWebClientId : '31kkeavn30400faua7h2kh08hs'
    }
});


class App extends Component {
  render() {
    return (
    <div className="App">
      <header className="App-header">
          <img src={owl} className="App-logo" alt="logo" />
          <p/ >
        <LoginForm />
        
      </header>
    </div>
    );
  }
}

export default App;
