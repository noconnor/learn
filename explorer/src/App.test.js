/*global expect*/
import React from 'react';
import App from './App';
import Menu from 'components/Explorer/Greetings'
import ExplorerForm from 'components/Explorer/Form'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { render, waitForElement } from 'react-testing-library';

// https://medium.freecodecamp.org/the-right-way-to-test-react-components-548a4736ab22
describe("App", () =>{
  
  let props;
  let liveApp;
  
  const app = () => {
    if (!liveApp) {
      liveApp = shallow(<App {...props} />);
    }
    return liveApp;
  }
  
  beforeEach(() => {
    props = {
      authState: undefined,
    };
    liveApp = undefined;
  });
  
  describe("when the app is not logged in", () => {
    it('component should not render', () => {
      expect(app().html()).toBeNull();
    });  
  });
  
  describe("when the app is logged in", () => {

    beforeEach(() => {
      props.authState = "signedIn";
    });

    // Example only, see case against https://engineering.ezcater.com/the-case-against-react-snapshot-testing
    it("should render as expected", () => {
      const tree = renderer.create(<App {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    
    // *** instead of snapshot *** 
    it("always renders a div", () => {
      const divs = app().find("div");
      expect(divs.length).toBeGreaterThan(0);
    });
    
    describe("the rendered div", () => {
      it("contains everything else that gets rendered", () => {
        const divs = app().find("div");
        const wrappingDiv = divs.first();
        expect(wrappingDiv.children()).toEqual(app().children());
      });
    });
    // *** end instead of snapshot *** 

    describe("the menu component", () => {
      it('should be rendered', () => {
        expect(app().find(Menu).length).toBe(1);
      });
      it('should inherit authState props', () => {
        const menu = app().find(Menu);
        expect(menu.props().authState).toBe(props.authState);
      });
      
    });
    
    describe("the explorer form component", () => {
      it('should be rendered', () => {
        expect(app().find(ExplorerForm).length).toBe(1);
      });
      it('should inherit authState props', () => {
        const explorer = app().find(ExplorerForm);
        expect(explorer.props().authState).toBe(props.authState);
      });
    });
    
    describe("the header element", () => {
      it('should be rendered', () => {
        expect(app().find("header").length).toBe(1);
      });
    });

    describe("the logo image element", () => {
      it('should be rendered', () => {
        expect(app().find("img.App-logo").length).toBe(1);
      });
    });
    
  });
  
});
