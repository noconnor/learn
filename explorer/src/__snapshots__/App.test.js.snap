// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App when the app is logged in should render as expected 1`] = `
<div
  className="App"
>
  <div
    className="menu"
  >
    <nav
      className="navbar navbar-inverse"
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <a
            className="navbar-brand"
            href="#brand"
          >
            Explorer
          </a>
          <button
            className="navbar-toggle collapsed"
            onClick={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
        >
          <ul
            className="nav navbar-nav navbar-right"
            role={null}
          >
            <li
              className=""
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                Sign Out
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="owl.png"
    />
    <p />
    <form>
      <div
        className="form-group has-feedback has-error"
      >
        <input
          className="form-control"
          id="formBasicText"
          onChange={[Function]}
          placeholder="Enter URL"
          type="text"
          value=""
        />
        <br />
        <span
          className="form-control-feedback glyphicon glyphicon-remove"
        />
        <button
          className="btn btn-primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Store
        </button>
      </div>
    </form>
  </header>
</div>
`;
